dydx/dydxPresenters/dydxPresenters/_v4/TakeProfitStopLoss/dydxTakeProfitStopLossViewPresenter.swift

```diff
@@ -46,25 +49,110 @@ private class dydxTakeProfitStopLossViewPresenter: HostedViewPresenter<dydxTakeP
+        viewModel?.takeProfitStopLossInputAreaViewModel?.takeProfitPriceInputViewModel?.value = dydxFormatter.shared.dollar(number: triggerOrdersInput?.takeProfitOrder?.price?.triggerPrice?.doubleValue)
+        viewModel?.takeProfitStopLossInputAreaViewModel?.gainInputViewModel?.value = dydxFormatter.shared.dollar(number: triggerOrdersInput?.takeProfitOrder?.price?.usdcDiff?.doubleValue)
+        viewModel?.takeProfitStopLossInputAreaViewModel?.stopLossPriceInputViewModel?.value = dydxFormatter.shared.dollar(number: triggerOrdersInput?.stopLossOrder?.price?.triggerPrice?.doubleValue)
+        viewModel?.takeProfitStopLossInputAreaViewModel?.lossInputViewModel?.value = dydxFormatter.shared.dollar(number: triggerOrdersInput?.stopLossOrder?.price?.usdcDiff?.doubleValue)
+    }
+
+    private func update(subaccountPositions: [SubaccountPosition], subaccountOrders: [SubaccountOrder]) {
+        // TODO: move this logic to abacus
+        let position = subaccountPositions.first { subaccountPosition in
+            subaccountPosition.id == marketId
+        }
+        let takeProfitOrders = subaccountOrders.filter { (order: SubaccountOrder) in
+            order.marketId == marketId && (order.type == .takeprofitmarket || order.type == .takeprofitlimit) && order.side.opposite == position?.side.current
+        }
+        let stopLossOrders = subaccountOrders.filter { (order: SubaccountOrder) in
+            order.marketId == marketId && (order.type == .stopmarket || order.type == .stoplimit) && order.side.opposite == position?.side.current
+        }
+
+        viewModel?.entryPrice = position?.entryPrice?.current?.doubleValue
+
+        viewModel?.takeProfitStopLossInputAreaViewModel?.numOpenTakeProfitOrders = takeProfitOrders.count
+        viewModel?.takeProfitStopLossInputAreaViewModel?.numOpenStopLossOrders = stopLossOrders.count
+
+        if takeProfitOrders.count == 1, let order = takeProfitOrders.first {
+            AbacusStateManager.shared.triggerOrders(input: order.size.description, type: .takeprofitordersize)
+            AbacusStateManager.shared.triggerOrders(input: order.type.rawValue, type: .takeprofitordertype)
+            AbacusStateManager.shared.triggerOrders(input: order.price.description, type: .takeprofitlimitprice)
+            AbacusStateManager.shared.triggerOrders(input: order.triggerPrice?.stringValue, type: .takeprofitprice)
+        }
+        if stopLossOrders.count == 1, let order = stopLossOrders.first {
+            AbacusStateManager.shared.triggerOrders(input: order.size.description, type: .stoplossordersize)
+            AbacusStateManager.shared.triggerOrders(input: order.type.rawValue, type: .stoplossordertype)
+            AbacusStateManager.shared.triggerOrders(input: order.price.description, type: .stoplosslimitprice)
+            AbacusStateManager.shared.triggerOrders(input: order.triggerPrice?.stringValue, type: .stoplossprice)
+        }
```

Comment:
this is per how mulan implemented for web. The flow is

1. FE starts to list for positions/orders/triggerOrdersInput from abacus
2. FE does the logic to filter relevant positions/orders
3. FE feeds the relevant SL/TP order(s) into abacus via `triggerOrders` function
4. Abacus processes the inputs, does calculations, updates the state
5. FE reflects new state of `triggerOrdersInput`

I want to move this into abacus. mulan and Jared say it will be difficult because orders and positions are isolated from each other in abacus currently.